bin/release-rewards.js

```diff
@@ -7,25 +7,46 @@ import * as SparkImpactEvaluator from '@filecoin-station/spark-impact-evaluator'
 import readline from 'node:readline/promises'
 import pRetry from 'p-retry'
 import beeper from 'beeper'
+import pMap from 'p-map'
 
 process.title = 'release-rewards'
 const { RPC_URL = 'https://api.node.glif.io/rpc/v1', WALLET_SEED } = process.env
 
+const provider = new ethers.JsonRpcProvider(RPC_URL)
+const ie = new ethers.Contract(SparkImpactEvaluator.ADDRESS, SparkImpactEvaluator.ABI, provider)
+
 const rawRewardsRes = await fetch('https://spark-rewards.fly.dev/scheduled-rewards')
 const rawRewards = await rawRewardsRes.json()
-const rewards = Object.entries(rawRewards)
+const unfilteredRewards = Object.entries(rawRewards)
   .map(([address, amount]) => ({
     address,
-    amount: BigInt(amount),
-    amountFIL: Number(amount) / 1e18
+    amount: BigInt(amount)
   }))
-  .filter(({ amount }) => amount > 0n)
+unfilteredRewards.sort((a, b) => Number(b.amount - a.amount))
+
+console.log(`Found ${unfilteredRewards.length} participants with spark-rewards scheduled rewards`)
+console.log('Filtering out participants with total scheduled rewards (spark-rewards + smart contract) below 0.1 FIL...')
+const rewards = []
+await pMap(
+  unfilteredRewards,
+  async ({ address, amount }, index) => {
+    if (index > 0 && index % 100 === 0) {
+      console.log(`${index}/${unfilteredRewards.length}`)
+    }
+    if (amount === 0) return
+    const totalScheduledRewards =
+      (await ie.rewardsScheduledFor(address)) + amount
+    if (totalScheduledRewards >= 0.1e18) {
```

Comment:
I think the following would be easier to read:

```suggestion
    if (totalScheduledRewards >= 0.1 * 1e18) {
```

Do you have any concerns about comparing a BigInt value against a float value and potentially losing precision?

A possible solution that avoids the problem:

```suggestion
    if (totalScheduledRewards >= /* 0.1 FIL */ 10n ** 18n / 10n) {
```

But maybe this is not an issue.


```
❯ node                                
Welcome to Node.js v20.11.1.
Type ".help" for more information.
> 10n ** 18n / 10n
100000000000000000n
> 0.1e18
100000000000000000
> 
```

Reply:
I don't have an opinion on this, as the results are the same and for me the current version is easiest to read. What do you prefer?

Reply:
I find `0.1 * 1e18` easier to read than `0.1e18` - I have to think a bit to decipher what `0.1e18` means.

Adding a code comment explaining the threshold is `0.1 FIL` is probably the best solution, as it does not require any knowledge about attoFIL being $1^{-18}$.

Anyhow, this is a detail not worth bikeshedding. Choose the option you prefer yourself!